### ISSUE TITLE
Error: shasum check failed 

### ISSUE THREAD

**Issue by Andis59** [saebekassebil/musicjson#4]
I tried to install musicjson on Windows 7 and I get "Error: shasum check failed" (see whole log file last). What is the problem and how do I fix it?
// andis

```
0 info it worked if it ends with ok
1 verbose cli [ 'node.exe',
1 verbose cli   'C:\\Chocolatey\\lib\\nodist.0.3.12-beta\\tools\\nodist-master\\bin\\node_modules\\npm\\cli.js',
1 verbose cli   'install',
1 verbose cli   '-g',
1 verbose cli   'musicjson' ]
2 info using npm@1.4.4
3 info using node@v0.10.26
4 verbose cache add [ 'musicjson', null ]
5 verbose cache add name=undefined spec="musicjson" args=["musicjson",null]
6 verbose parsed url { protocol: null,
6 verbose parsed url   slashes: null,
6 verbose parsed url   auth: null,
6 verbose parsed url   host: null,
6 verbose parsed url   port: null,
6 verbose parsed url   hostname: null,
6 verbose parsed url   hash: null,
6 verbose parsed url   search: null,
6 verbose parsed url   query: null,
6 verbose parsed url   pathname: 'musicjson',
6 verbose parsed url   path: 'musicjson',
6 verbose parsed url   href: 'musicjson' }
7 silly lockFile 3c5e0d18-musicjson musicjson
8 verbose lock musicjson C:\Users\ame\AppData\Roaming\npm-cache\3c5e0d18-musicjson.lock
9 silly lockFile 3c5e0d18-musicjson musicjson
10 silly lockFile 3c5e0d18-musicjson musicjson
11 verbose addNamed [ 'musicjson', '' ]
12 verbose addNamed [ null, '*' ]
13 silly lockFile 9cee5574-musicjson musicjson@
14 verbose lock musicjson@ C:\Users\ame\AppData\Roaming\npm-cache\9cee5574-musicjson.lock
15 silly addNameRange { name: 'musicjson', range: '*', hasData: false }
16 verbose url raw musicjson
17 verbose url resolving [ 'https://registry.npmjs.org/', './musicjson' ]
18 verbose url resolved https://registry.npmjs.org/musicjson
19 info trying registry request attempt 1 at 17:26:52
20 verbose etag "KDTPBWB3W1YJQ1GBB1Z6E6QM"
21 http GET https://registry.npmjs.org/musicjson
22 http 304 https://registry.npmjs.org/musicjson
23 silly registry.get cb [ 304,
23 silly registry.get   { date: 'Tue, 11 Mar 2014 16:26:53 GMT',
23 silly registry.get     server: 'Apache',
23 silly registry.get     via: '1.1 varnish',
23 silly registry.get     'last-modified': 'Tue, 11 Mar 2014 16:26:53 GMT',
23 silly registry.get     'cache-control': 'max-age=1',
23 silly registry.get     etag: '"KDTPBWB3W1YJQ1GBB1Z6E6QM"',
23 silly registry.get     'x-served-by': 'cache-am70-AMS',
23 silly registry.get     'x-cache': 'MISS',
23 silly registry.get     'x-cache-hits': '0',
23 silly registry.get     'x-timer': 'S1394555213.084800243,VS0,VE180',
23 silly registry.get     vary: 'Accept',
23 silly registry.get     'content-length': '0',
23 silly registry.get     'keep-alive': 'timeout=10, max=50',
23 silly registry.get     connection: 'Keep-Alive' } ]
24 verbose etag musicjson from cache
25 silly addNameRange number 2 { name: 'musicjson', range: '*', hasData: true }
26 silly addNameRange versions [ 'musicjson',
26 silly addNameRange   [ '0.1.0', '0.1.1', '0.1.2', '0.1.3', '0.1.5', '0.1.6', '0.2.0' ] ]
27 verbose addNamed [ 'musicjson', '0.2.0' ]
28 verbose addNamed [ '0.2.0', '0.2.0' ]
29 silly lockFile e7f0bc82-musicjson-0-2-0 musicjson@0.2.0
30 verbose lock musicjson@0.2.0 C:\Users\ame\AppData\Roaming\npm-cache\e7f0bc82-musicjson-0-2-0.lock
31 silly lockFile 012f8cf3-rg-musicjson-musicjson-0-2-0-tgz https://registry.npmjs.org/musicjson/-/musicjson-0.2.0.tgz
32 verbose lock https://registry.npmjs.org/musicjson/-/musicjson-0.2.0.tgz C:\Users\ame\AppData\Roaming\npm-cache\012f8cf3-rg-musicjson-musicjson-0-2-0-tgz.lock
33 verbose addRemoteTarball [ 'https://registry.npmjs.org/musicjson/-/musicjson-0.2.0.tgz',
33 verbose addRemoteTarball   '37f101a58f20c94de20e7a14ec4d8e9b36f06cd3' ]
34 info retry fetch attempt 1 at 17:26:52
35 verbose fetch to= C:\Users\ame\AppData\Local\Temp\npm-10224-uQUeGqZq\1394555212565-0.4306404253002256\tmp.tgz
36 http GET https://registry.npmjs.org/musicjson/-/musicjson-0.2.0.tgz
37 http 200 https://registry.npmjs.org/musicjson/-/musicjson-0.2.0.tgz
38 silly lockFile 012f8cf3-rg-musicjson-musicjson-0-2-0-tgz https://registry.npmjs.org/musicjson/-/musicjson-0.2.0.tgz
39 silly lockFile 012f8cf3-rg-musicjson-musicjson-0-2-0-tgz https://registry.npmjs.org/musicjson/-/musicjson-0.2.0.tgz
40 silly lockFile e7f0bc82-musicjson-0-2-0 musicjson@0.2.0
41 silly lockFile e7f0bc82-musicjson-0-2-0 musicjson@0.2.0
42 silly lockFile 9cee5574-musicjson musicjson@
43 silly lockFile 9cee5574-musicjson musicjson@
44 error Error: shasum check failed for C:\Users\ame\AppData\Local\Temp\npm-10224-uQUeGqZq\1394555212565-0.4306404253002256\tmp.tgz
44 error Expected: 37f101a58f20c94de20e7a14ec4d8e9b36f06cd3
44 error Actual:   317cde3fb70926f4bddbc8307c773012d863fe20
44 error From:     https://registry.npmjs.org/musicjson/-/musicjson-0.2.0.tgz
44 error     at C:\Chocolatey\lib\nodist.0.3.12-beta\tools\nodist-master\bin\node_modules\npm\node_modules\sha\index.js:38:8
44 error     at ReadStream.<anonymous> (C:\Chocolatey\lib\nodist.0.3.12-beta\tools\nodist-master\bin\node_modules\npm\node_modules\sha\index.js:85:7)
44 error     at ReadStream.EventEmitter.emit (events.js:117:20)
44 error     at _stream_readable.js:920:16
44 error     at process._tickCallback (node.js:415:13)
45 error If you need help, you may report this *entire* log,
45 error including the npm and node versions, at:
45 error     <http://github.com/npm/npm/issues>
46 error System Windows_NT 6.1.7601
47 error command "node.exe" "C:\\Chocolatey\\lib\\nodist.0.3.12-beta\\tools\\nodist-master\\bin\\node_modules\\npm\\cli.js" "install" "-g" "musicjson"
48 error cwd C:\Users\ame
49 error node -v v0.10.26
50 error npm -v 1.4.4
51 verbose exit [ 1, true ]
```

**Comment by Andis59** [comment]
I tested on Lubuntu 13.10 and got the same error.
// andis
---- log file ----

```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/nodejs', '/usr/bin/npm', 'install', '-g', 'musicjson' ]
2 info using npm@1.2.18
3 info using node@v0.10.15
4 verbose read json /usr/local/lib/package.json
5 verbose read json /usr/local/lib/package.json
6 verbose cache add [ 'musicjson', null ]
7 verbose cache add name=undefined spec="musicjson" args=["musicjson",null]
8 verbose parsed url { protocol: null,
8 verbose parsed url   slashes: null,
8 verbose parsed url   auth: null,
8 verbose parsed url   host: null,
8 verbose parsed url   port: null,
8 verbose parsed url   hostname: null,
8 verbose parsed url   hash: null,
8 verbose parsed url   search: null,
8 verbose parsed url   query: null,
8 verbose parsed url   pathname: 'musicjson',
8 verbose parsed url   path: 'musicjson',
8 verbose parsed url   href: 'musicjson' }
9 silly lockFile 3c5e0d18-musicjson musicjson
10 verbose lock musicjson /home/ame/.npm/3c5e0d18-musicjson.lock
11 silly lockFile 3c5e0d18-musicjson musicjson
12 silly lockFile 3c5e0d18-musicjson musicjson
13 verbose addNamed [ 'musicjson', '' ]
14 verbose addNamed [ null, '*' ]
15 silly lockFile 9cee5574-musicjson musicjson@
16 verbose lock musicjson@ /home/ame/.npm/9cee5574-musicjson.lock
17 silly addNameRange { name: 'musicjson', range: '*', hasData: false }
18 verbose url raw musicjson
19 verbose url resolving [ 'https://registry.npmjs.org/', './musicjson' ]
20 verbose url resolved https://registry.npmjs.org/musicjson
21 info trying registry request attempt 1 at 19:29:17
22 http GET https://registry.npmjs.org/musicjson
23 http 200 https://registry.npmjs.org/musicjson
24 silly registry.get cb [ 200,
24 silly registry.get   { date: 'Tue, 11 Mar 2014 18:29:19 GMT',
24 silly registry.get     server: 'CouchDB/1.5.0 (Erlang OTP/R16B)',
24 silly registry.get     etag: '"KDTPBWB3W1YJQ1GBB1Z6E6QM"',
24 silly registry.get     'content-type': 'application/json',
24 silly registry.get     via: '1.1 varnish',
24 silly registry.get     'cache-control': 'max-age=1',
24 silly registry.get     'content-length': '9759',
24 silly registry.get     'accept-ranges': 'bytes',
24 silly registry.get     age: '0',
24 silly registry.get     'x-served-by': 'cache-v37-ASH, cache-fra1223-FRA',
24 silly registry.get     'x-cache': 'MISS, MISS',
24 silly registry.get     'x-cache-hits': '0, 0',
24 silly registry.get     'x-timer': 'S1394562559.703306675,VS0,VS355,VE451,VE188',
24 silly registry.get     vary: 'Accept',
24 silly registry.get     'keep-alive': 'timeout=10, max=50',
24 silly registry.get     connection: 'Keep-Alive' } ]
25 silly addNameRange number 2 { name: 'musicjson', range: '*', hasData: true }
26 silly addNameRange versions [ 'musicjson',
26 silly addNameRange   [ '0.1.0', '0.1.1', '0.1.2', '0.1.3', '0.1.5', '0.1.6', '0.2.0' ] ]
27 verbose addNamed [ 'musicjson', '0.2.0' ]
28 verbose addNamed [ '0.2.0', '0.2.0' ]
29 silly lockFile e7f0bc82-musicjson-0-2-0 musicjson@0.2.0
30 verbose lock musicjson@0.2.0 /home/ame/.npm/e7f0bc82-musicjson-0-2-0.lock
31 silly lockFile 012f8cf3-rg-musicjson-musicjson-0-2-0-tgz https://registry.npmjs.org/musicjson/-/musicjson-0.2.0.tgz
32 verbose lock https://registry.npmjs.org/musicjson/-/musicjson-0.2.0.tgz /home/ame/.npm/012f8cf3-rg-musicjson-musicjson-0-2-0-tgz.lock
33 verbose addRemoteTarball [ 'https://registry.npmjs.org/musicjson/-/musicjson-0.2.0.tgz',
33 verbose addRemoteTarball   '37f101a58f20c94de20e7a14ec4d8e9b36f06cd3' ]
34 info retry fetch attempt 1 at 19:29:20
35 verbose fetch to= /home/ame/tmp/npm-7734/1394562560010-0.03178174560889602/tmp.tgz
36 http GET https://registry.npmjs.org/musicjson/-/musicjson-0.2.0.tgz
37 http 200 https://registry.npmjs.org/musicjson/-/musicjson-0.2.0.tgz
38 silly shasum updated bytes 4819
39 info shasum 317cde3fb70926f4bddbc8307c773012d863fe20
39 info shasum /home/ame/tmp/npm-7734/1394562560010-0.03178174560889602/tmp.tgz
40 silly lockFile 012f8cf3-rg-musicjson-musicjson-0-2-0-tgz https://registry.npmjs.org/musicjson/-/musicjson-0.2.0.tgz
41 silly lockFile 012f8cf3-rg-musicjson-musicjson-0-2-0-tgz https://registry.npmjs.org/musicjson/-/musicjson-0.2.0.tgz
42 silly lockFile e7f0bc82-musicjson-0-2-0 musicjson@0.2.0
43 silly lockFile e7f0bc82-musicjson-0-2-0 musicjson@0.2.0
44 silly lockFile 9cee5574-musicjson musicjson@
45 silly lockFile 9cee5574-musicjson musicjson@
46 error Error: shasum check failed for /home/ame/tmp/npm-7734/1394562560010-0.03178174560889602/tmp.tgz
46 error Expected: 37f101a58f20c94de20e7a14ec4d8e9b36f06cd3
46 error Actual:   317cde3fb70926f4bddbc8307c773012d863fe20
46 error     at /usr/share/npm/lib/utils/sha.js:29:20
46 error     at ReadStream.<anonymous> (/usr/share/npm/lib/utils/sha.js:49:5)
46 error     at ReadStream.EventEmitter.emit (events.js:117:20)
46 error     at _stream_readable.js:910:16
46 error     at process._tickCallback (node.js:415:13)
47 error If you need help, you may report this log at:
47 error     <http://bugs.debian.org/npm>
47 error or use
47 error     reportbug --attach /home/ame/npm-debug.log npm
48 error System Linux 3.11.0-15-generic
49 error command "/usr/bin/nodejs" "/usr/bin/npm" "install" "-g" "musicjson"
50 error cwd /home/ame
51 error node -v v0.10.15
52 error npm -v 1.2.18
53 verbose exit [ 1, true ]
```

**Comment by saebekassebil** [comment]
Hey @Andis59. This has most probably something to do with your npm distribution or/and your network connection. Could you try this from a different network as well? If you get the same error I suggest you report it at [npm's issue tracker](https://github.com/npm/npm/issues)

**Comment by Andis59** [comment]
I have tried from a different network and I get the same error.
If I install some other package it works. Well, anyway for the two I have tried.
Where is the expected check sum stored?

**Comment by saebekassebil** [comment]
I cleaned my cache and tried to install again, and got the same failure as you.

I think this might have something to do with npm's recent wrecking of forced updates. I've probably done one before they enforced the "no-forced updates" paradigm, but still I think that might be the cause of trouble.

I've bumped the version number and re-published, and I can now download as expected. Try now (and make sure it installs `v0.2.1` and not `v0.2.0`)
